app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,273 @@
+      def to_inventory_collection
+        if @properties[:model_class].nil? && !@options[:without_model_class]
+          raise MissingModelClassError
+        end
+
+        ::ManagerRefresh::InventoryCollection.new(to_hash)
+      end
+
+      #
+      # Missing method
+      #   - add_some_property(value)
+      # converted to:
+      #   - add_properties(:some_property => value)
+      #
+      def method_missing(method_name, *arguments, &block)
+        if method_name.to_s.starts_with?('add_')
+          add_properties(
+            method_name.to_s.gsub('add_', '').to_sym => arguments[0]
+          )
+        else
+          super
+        end
+      end
+
+      def respond_to_missing?(method_name, _include_private = false)
+        method_name.to_s.starts_with?('add_')
+      end
+
+      # Merges @properties
+      # @see ManagerRefresh::InventoryCollection.initialize for list of properties
+      #
+      # @param props [Hash]
+      # @param mode [Symbol] :overwrite | :if_missing
+      def add_properties(props = {}, mode = :overwrite)
+        @properties = merge_hashes(@properties, props, mode)
+      end
+
+      # Adds inventory object attributes (part of @properties)
+      def add_inventory_attributes(array)
+        @inventory_object_attributes += (array || [])
+      end
+
+      # Removes specified inventory object attributes
+      def remove_inventory_attributes(array)
+        @inventory_object_attributes -= (array || [])
+      end
+
+      # Clears all inventory object attributes
+      def clear_inventory_attributes!
+        @inventory_object_attributes = []
+      end
+
+      # Adds key/values to builder params (part of @properties)
+      def add_builder_params(params = {}, mode = :overwrite)
+        @builder_params = merge_hashes(@builder_params, params, mode)
+      end
+
+      # Evaluates lambda blocks
+      def evaluate_lambdas!(persister)
+        evaluate_builder_params_lambdas!(persister)
+      end
+
+      # Adds key/values to dependency_attributes (part of @properties)
+      def add_dependency_attributes(attrs = {}, mode = :overwrite)
+        @dependency_attributes = merge_hashes(@dependency_attributes, attrs, mode)
+      end
+
+      # Returns whole InventoryCollection properties
+      def to_hash
+        @properties.merge(
+          :inventory_object_attributes => @inventory_object_attributes,
+          :builder_params              => @builder_params,
+          :dependency_attributes       => @dependency_attributes
+        )
+      end
+
+      # Adds parent to InventoryCollection if not set before
+      # Can be disabled by :auto_missing_parent => false
+      def add_parent_if_missing(manager, targeted = false)
+        return unless @options[:auto_missing_parent]
+
+        if manager.present?
+          if targeted && network_manager_collections? && manager.respond_to?(:network_manager)
+            add_properties({:parent => manager.network_manager}, :if_missing)
```

Comment:
Shouldn't this be something that we fix explicitly in the inventory collections that are missing a parent?  I think this magic auto-guessing can lead to issues.

Reply:
This is also 1:1 refactoring:
https://github.com/slemrmartin/manageiq/blob/6971ccbeca86576f1fc866fe81381de1510195d9/app/models/manager_refresh/inventory/persister.rb#L130
https://github.com/slemrmartin/manageiq/blob/6971ccbeca86576f1fc866fe81381de1510195d9/app/models/manager_refresh/inventory/persister.rb#L60

Reply:
for network manager's collections it is 1:1 too, but it wasn't extracted to one place before. 
So it fits actual state.

But I doesn't have enough knowledge yet to know if it should lead to problems or not in the future, 
 so if you want, I can move it from builder to concrete persisters.

Should I?

Reply:
yeah, lets delete this and define it explicitly in specific persistors (if we define it in the shared options, it will not make the code ugly )